### PR TITLE
Validate the eval dataset in advance.

### LINKED ISSUE #33721: validation of the eval dataset should be done in advance
### System Info

I found that the validation of the eval dataset should be done in advance. It only prompted that the validation set was empty after I had trained for half an hour and completed one epoch.The code is as follows.I set                         evaluation_strategy="epoch" in train_args,  but I don't set eval dataset in Trainer init.

`from transformers import TrainingArguments

train_args = TrainingArguments(output_dir="./checkpoints",     
                               per_device_train_batch_size=2, 
                               per_device_eval_batch_size=1,  
                               logging_steps=1,                
                               **evaluation_strategy="epoch",**    
                               save_strategy="epoch",          
                               save_total_limit=3,            
                               learning_rate=2e-5,           
                               weight_decay=0.01,
                               num_train_epochs=30,
                               dataloader_drop_last=True,
                               metric_for_best_model="f1",   
                               load_best_model_at_end=True
                               )  

from transformers import DataCollatorWithPadding, Trainer

trainer = Trainer(model=model, 
                  args=train_args, 
                  train_dataset=process_dataset["train"], 
                  **eval_dataset=None,** 
                  data_collator=collate_fn)

`

transformers version:4.45.0
python version:3.10
paltform:ubuntu

@muellerzr @Sunm

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. set evaluation_strategy="epoch" in train_args
2. set eval dataset=None in Trainer init
3. run Trainer.run()

### Expected behavior

validation of the eval dataset should be done in advance

### PR DESCRIPTION
**Pull request by jackyjinjing**
# What does this PR do?
Validate whether the eval dataset is empty in advance to avoid wasting the time of training for one epoch only to discover this issue.

Fixes # (issue)
fix #33721

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@muellerzr and @SunMarc

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -401,6 +401,8 @@ def __init__(
                     " boolean argument which will be triggered after the last batch of the eval set to signal that the"
                     " summary statistics should be returned by the function."
                 )
+        if args.eval_strategy is not None and args.eval_strategy != "no" and eval_dataset is None:
+            raise ValueError("Trainer: evaluation requires an eval_dataset.")
```
Comment (SunMarc) [inline src/transformers/trainer.py:405]:
```suggestion
        if args.eval_strategy is not None and args.eval_strategy != "no" and eval_dataset is None:
            raise ValueError(f"You have set `args.eval_strategy` to {args.eval_strategy} but you didn't pass an `eval_dataset` to `Trainer`. Either set `args.eval_strategy` to `no` or pass an `eval_dataset`. ")
```

**Review by SunMarc — APPROVED** [review]
Thanks for adding this @jackyjinjing ! Just a nit

**Review by SunMarc — APPROVED** [review]
Thanks !

**Review by LysandreJik — APPROVED** [review]
Awesome, thanks!
